packages/flutter_tools/test/general.shard/build_system/targets/dart_plugin_registrant_test.dart

```diff
@@ -0,0 +1,226 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'dart:io';
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_system/build_system.dart';
+import 'package:flutter_tools/src/build_system/targets/dart_plugin_registrant.dart';
+import 'package:flutter_tools/src/project.dart';
+
+import '../../../src/common.dart';
+import '../../../src/context.dart';
+
+const String _kSamplePackageJson = '''
+{
+  "configVersion": 2,
+  "packages": [
+    {
+      "name": "path_provider_linux",
+      "rootUri": "/path_provider_linux",
+      "packageUri": "lib/",
+      "languageVersion": "2.12"
+    }
+  ]
+}
+''';
+
+const String _kSamplePackagesFile = '''
+path_provider_linux:/path_provider_linux/lib/
+''';
+
+const String _kSamplePackageJsonWindows = '''
+{
+  "configVersion": 2,
+  "packages": [
+    {
+      "name": "path_provider_linux",
+      "rootUri": "file:///C:/path_provider_linux",
+      "packageUri": "lib/",
+      "languageVersion": "2.12"
+    }
+  ]
+}
+''';
+
+const String _kSamplePackagesFileWindows = '''
+path_provider_linux:file:///C:/path_provider_linux/lib/
+''';
+
+const String _kSamplePubspecFile = '''
+name: path_provider_example
+description: Demonstrates how to use the path_provider plugin.
+
+dependencies:
+  flutter:
+    sdk: flutter
+  path_provider_linux: 1.0.0
+''';
+
+const String _kLinuxRegistrant =
```

Comment:
if you use a multiline string, then you don't need to end each line with a `\n`, which will make this much easier to read.

Reply:
This is the output of the test. I'd prefer to keep it like this since it's easier to update/see diffs, etc... This format is also used in other tests.

Reply:
I'm not sure why this would be easier to update than a multiline string. e.g.:

```dart
const String _kLinuxRegistrant = """
//
// Generated file. Do not edit.
//

// @dart = 2.12

import '\' as entrypoint;
import 'dart:io'; // flutter_ignore: dart_io_import.
import 'package:path_provider_linux/path_provider_linux.dart';
...
""";
```

Also the chosen entry point of this golden seems suspicious - it might be missing a configured target?

Reply:
This is how the test runner suggests edits: https://gist.github.com/blasten/e3c6ac2433e29837a77308797a2b17a9

Similar to how scuba or other baseline-like tests do it, I just copied the suggested output.